### PR TITLE
Expose RMSProp optimizer.

### PR DESCRIPTION
**Pull request by qingqing01**
Fix https://github.com/PaddlePaddle/Paddle/issues/9246

### CODE REVIEW

File: python/paddle/fluid/optimizer.py [hunk 1]
```diff
@@ -664,6 +664,113 @@ def _append_optimize_op(self, block, param_and_grad):
         return adadelta_op
 
 
+class RMSPropOptimizer(Optimizer):
+    """
+    Root Mean Squared Propagation (RMSProp) is an unpublished, adaptive learning
+    rate method. The original slides that proposed Rmsprop: Slide 29 of
```
Comment (pkuyym) [inline python/paddle/fluid/optimizer.py]: slides that proposed Rmsprop --> slides proposed Rmsprop

File: python/paddle/fluid/optimizer.py [hunk 2]
```diff
@@ -664,6 +664,113 @@ def _append_optimize_op(self, block, param_and_grad):
...
+    The first equation calculates moving average of the squared gradient for
+    each weight. Then dividing the gradient by :math: `sqrt{v(w,t)}`.
+
+    In some cases, adding a momentum term :math: `\\beta` is beneficial.
+    In our implementation, Nesterov momentum is used:
+
+    ..  math::
+
+        r(w, t) & = \\rho r(w, t-1) + (1 - \\rho)(\\nabla Q_{i}(w))^2 \\\\
+        v(w, t) & = \\beta v(w, t-1) + \\frac{\\eta} {\\sqrt{v(w,t) + \\epsilon}} \\nabla Q_{i}(w)
```
Comment (pkuyym) [inline python/paddle/fluid/optimizer.py]: Seems exceeded 80 columns.

File: python/paddle/fluid/optimizer.py [hunk 3]
```diff
@@ -664,6 +664,113 @@ def _append_optimize_op(self, block, param_and_grad):
...
+    and so on. :math: `beta` is the momentum term. :math: `\\epsilon` is a
+    smoothing term to avoids division by zero, usually set somewhere in range
+    from 1e-4 to 1e-8.
+
+
+    Args:
+        learning_rate(float): global leraning rate.
+        rho(float): rho is :math: `\\rho` in equation, set 0.95 by default.
+        epsilon(float): :math: `\\epsilon` in equation is smoothing term to
+            avoids division by zero, set 1e-6 by default.
```
Comment (pkuyym) [inline python/paddle/fluid/optimizer.py]: avoids --> avoid

File: python/paddle/fluid/optimizer.py [hunk 4]
```diff
@@ -664,6 +664,113 @@ def _append_optimize_op(self, block, param_and_grad):
...
+        momentum(float): :math: `\\beta` in equation is the momentum term,
+            set 0.0 by default.
+    
+
+    Examples:
```
Comment (pkuyym) [inline python/paddle/fluid/optimizer.py]: Add a new section `Raises` is better.

File: python/paddle/fluid/optimizer.py [hunk 5]
```diff
@@ -664,6 +664,113 @@ def _append_optimize_op(self, block, param_and_grad):
...
+        if epsilon is None:
+            raise ValueError("epsilon is not set.")
+
+        self.type = "rmsprop"
+        self._rho = rho
+        self._epsilon = epsilon
+        self._momentum = momentum
+
+    def _create_accumulators(self, block, parameters):
+        assert isinstance(block, framework.Block)
```
Comment (wanghaoshuang) [inline python/paddle/fluid/optimizer.py]: 这里是不是也需要改成raise，以免assert在非debug模式下失效？

File: python/paddle/fluid/optimizer.py (hunk 1 above)
Comment (qingqing01) [reply python/paddle/fluid/optimizer.py]: Done.

File: python/paddle/fluid/optimizer.py (hunk 2 above)
Comment (qingqing01) [reply python/paddle/fluid/optimizer.py]: Done.

File: python/paddle/fluid/optimizer.py (hunk 3 above)
Comment (qingqing01) [reply python/paddle/fluid/optimizer.py]: Done.

File: python/paddle/fluid/optimizer.py (hunk 4 above)
Comment (qingqing01) [reply python/paddle/fluid/optimizer.py]: Added.

File: python/paddle/fluid/optimizer.py (hunk 5 above)
Comment (qingqing01) [reply python/paddle/fluid/optimizer.py]: Done.

**Review by pkuyym — APPROVED** [review]
LGTM

**Pull request merged**